project/api/views.py

```diff
@@ -20,7 +22,16 @@ class TerritoryViewSet(viewsets.ModelViewSet):
     Viewset for the Territory model
     """
     permission_classes = (permissions.IsAuthenticatedOrReadOnly,)
-    queryset = Territory.objects.all()
     serializer_class = TerritorySerializer
 
+    queryset = Territory.objects.all()
+
+    def get_queryset(self):
+        bounds = self.request.query_params.get('bounds', None)
+        if bounds is not None:
+            geom = Polygon(make_tuple(bounds), srid=4326)
+            self.queryset = Territory.objects.filter(geo__within=geom)
```

Comment:
I didn't understand `within` in the docs https://docs.djangoproject.com/en/2.0/ref/contrib/gis/geoquerysets/. Isn't `bboverlaps` what we want? (No experience in this, just asking).

Reply:
Yeah I think you're right actually, I don't have any experience in this either lol